broker/setup_broker/setup_broker.sh

```diff
@@ -36,34 +36,80 @@ if [ "$continue_with_setup" != "y" ]; then
     exit
 fi
 
+# function used to define GCP resources; appends testid if needed
+define_GCP_resources() {
+    local base_name="$1"
+    local testid_suffix=""
+
+    if [ "$testid" != "False" ]; then
+        if [ "$base_name" = "$survey" ]; then
+            testid_suffix="_${testid}"  # complies with BigQuery naming conventions
+        else
+            testid_suffix="-${testid}"
+        fi
+    fi
+
+    echo "${base_name}${testid_suffix}"
+}
+
 #--- GCP resources used directly in this script
-broker_bucket="${PROJECT_ID}-${survey}-broker_files"
-bq_dataset="${survey}"
-# use test resources, if requested
-# (there must be a better way to do this)
-if [ "$testid" != "False" ]; then
-    broker_bucket="${broker_bucket}-${testid}"
-    bq_dataset="${bq_dataset}_${testid}"
-fi
+broker_bucket=$(define_GCP_resources "${PROJECT_ID}-${survey}-broker_files")
+bq_dataset=$(define_GCP_resources "${survey}")
+# topics and subscriptions involved in writing alert data to BigQuery
+topic_bigquery_import=$(define_GCP_resources "${survey}-bigquery-import")
+subscription_bigquery_import="${topic_bigquery_import}" # BigQuery subscription
+deadletter_topic_bigquery_import=$(define_GCP_resources "${survey}-bigquery-import-deadletter")
+deadletter_subscription_bigquery_import="${deadletter_topic_bigquery_import}"
+
+alerts_table="alerts_${versiontag}"
+
+# function used to create (or delete) GCP resources
+manage_resources() {
+    local mode="$1"  # setup or teardown
+    local environment_type="production"
+
+    if [ "$testid" != "False" ]; then
+        environment_type="testing"
+    fi
 
+    if [ "$mode" = "setup" ]; then
+        # setup resources
```

Comment:
You're creating/deleting the new Pub/Sub resources directly in this script but all the others are handled in setup_gcp.py. That's fine (we probably want to migrate away from that python script), but add a comment stating that this is the case.